relayer/app/cursor/cursors.go

```diff
@@ -0,0 +1,133 @@
+package cursor
+
+import (
+	"context"
+	"time"
+
+	"github.com/omni-network/omni/lib/errors"
+	"github.com/omni-network/omni/lib/expbackoff"
+	"github.com/omni-network/omni/lib/log"
+	"github.com/omni-network/omni/lib/netconf"
+	"github.com/omni-network/omni/lib/xchain"
+
+	db "github.com/cosmos/cosmos-db"
+)
+
+// Cursors implements operations on the persisted cursors for network.
+type Cursors struct {
+	cursors         CursorTable
+	confirmInterval time.Duration
+	xProvider       xchain.Provider
+}
+
+func NewCursors(
+	db db.DB,
+	xProvider xchain.Provider,
+	confirmInterval time.Duration,
+) (*Cursors, error) {
+	cursors, err := NewCursorsTable(db)
+	if err != nil {
+		return nil, err
+	}
+
+	return &Cursors{
+		cursors:         cursors,
+		confirmInterval: confirmInterval,
+		xProvider:       xProvider,
+	}, nil
+}
+
+// ConfirmedOffset loads confirmed cursor offset for the chain version and destination chain ID.
+// If a cursor doesn't exist a 0, false is returned.
+func (c *Cursors) ConfirmedOffset(
+	ctx context.Context,
+	chainVer xchain.ChainVersion,
+	destChainID uint64,
+) (uint64, bool, error) {
+	return newStreamCursors(chainVer.ID, destChainID, chainVer.ConfLevel, c.cursors, c.xProvider, "", "").
```

Comment:
Can you elaborate. If I understand you right you would completely remove the StreamCursors type and create functions that would all be passed in src chain, dest chain id, cursors table, xprovider, source name, destination name? That feels like they all operate on same data that's why it felt to me like they belong on an object owning the data and manipulating it. I can convert into functions if you think that's more appropriate tho

Reply:
yup, rather create a `stream` struct, which is the data:
```
type stream struct {
  SrcChain
  ConfLevel
  DestChain
}
```

then pass that into functions like `getConfirmed`
```
func getConfirmed(ctx, cursors, stream)
```

This splits logic from data. Instead of OO style which combines them.

Reply:
ok will change into this style